File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function(nums, k) {
    let count = 0;
    let currentSum = 0;
    let start = 0;

    for (let i = 0; i < nums.length; i++) {
        currentSum += nums[i];

        while (currentSum * (i - start + 1) >= k) {
            currentSum -= nums[start];
            start++;
        }
        count += (i - start + 1)
    }
    return count;
};

File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        count = 0
        currentSum = 0
        start = 0

        for end in range(len(nums)):
            currentSum += nums[end]
            while currentSum * (end - start + 1) >= k:
                currentSum -= nums[start]
                start += 1
            count += end - start + 1
        
        return count